src/components/ObsEdit/DeleteObservationDialog.js

```diff
@@ -19,25 +21,36 @@ const DeleteObservationDialog = ( {
   hideDialog
 }: Props ): Node => {
   const {
-    deleteCurrentObservation
+    deleteLocalObservation,
+    currentObservation
   } = useContext( ObsEditContext );
-  const deleteObservation = async ( ) => {
-    deleteCurrentObservation( );
-    hideDialog( );
-  };
+  const navigation = useNavigation( );
+  const { uuid } = currentObservation;
+
+  const deleteObservationMutation = useAuthenticatedMutation(
+    ( params, optsWithAuth ) => deleteObservation( params, optsWithAuth ),
+    {
+      onSuccess: ( ) => {
+        deleteLocalObservation( uuid );
+        hideDialog( );
+        navigation.navigate( "ObsList" );
+      }
+    }
+  );
 
   return (
     <Portal>
       <Dialog visible={deleteDialogVisible} onDismiss={hideDialog}>
         <Dialog.Content>
-          <Paragraph>{t( "Are-you-sure" )}</Paragraph>
+          <Text>{t( "Are-you-sure" )}</Text>
         </Dialog.Content>
         <Dialog.Actions>
-          <Button onPress={hideDialog} text={t( "Cancel" )} level="primary" />
+          <Button onPress={hideDialog} text={t( "Cancel" )} level="primary" className="m-0.5" />
```

Comment:
For some reason the text in this button isn't centered. I looked into it a little bit and didn't see an obvious solution. The `style` prop seems to get populated with mysterious padding from... somewhere, maybe some react-native-paper magic.